graphdatascience/graph_data_science.py

```diff
@@ -32,7 +32,31 @@ def __init__(
         auth: Optional[Tuple[str, str]] = None,
         aura_ds: bool = False,
         arrow: bool = True,
+        arrow_disable_server_verification: bool = True,
+        arrow_tls_root_certs: Optional[bytes] = None,
     ):
+        """
+        Parameters
+        ----------
+        endpoint : Union[str, Driver, QueryRunner]
+            The Neo4j endpoint to connect to
+        auth : Optional[Tuple[str, str]], default None
+            A username, password pair for database authentication.
+        aura_ds : bool, default False
+            A flag that indicates that that the client is used to connect
+            to a Neo4j Aura instance.
+        arrow : bool, default True
+            A flag that indicates that the client should use Apache Arrow
+            for data streaming if it is available on the server.
+        arrow_disable_server_verification : bool, default True
+            A flag that indicates that, if the flight client is connecting with
+            TLS, that it skips server verification. If this is enabled, all
+            other TLS settings are overridden.
+        arrow_tls_root_certs : Optional[bytes], default None
```

Comment:
A better UX would be falling back to the OS root CA bundle (I used python certifi to find it in my tests). But that's up to you, of course.

Reply:
Arrow / grpc already has a mechanism for setting the default value: https://grpc.github.io/grpc/cpp/structgrpc_1_1_ssl_credentials_options.html#a0ce1730020e18d04b6af48c88e069869

If we always set the value in the GDS client, changing that env var won't have an effect, i.e. users that know about that will be surprised. Maybe it's also an option for Aura to not handle the default in the client but setting the env variable instead?

Reply:
I mean, wouldn't a typical user be overwhelmed with finding where the root CA bundle lives on their machine? And neo4j driver picks it up automatically for bolt encrypted connection. But at the end, these are your users so 🤷